### PR TITLE
Documentation Improvement

### PR DESCRIPTION
**Pull request by masoudabedinifar**
Dear @rmndrs89  and @JuliusWelzel ,

Could you please review the improvements in the documentation? If they meet your requirements, could you please merge the branch into the main branch?
Thank you!

Best regards,
Masoud

### CODE REVIEW

**Comment by JuliusWelzel** [comment]
Hi,

 a few minor changes.
 
 - [ ] Note box https://github.com/neurogeriatricskiel/KielMAT/blob/c366fd834ffe8fd60e411ef85fef849e979cfed8/docs/index.md?plain=1#L152
 Does not render in the documentation properly
 
 - [ ] The tables on the index page should be markdown formatted.
 
 - [ ] Units For the units, please provide the accompanying channel types: https://github.com/neurogeriatricskiel/KielMAT/blob/c366fd834ffe8fd60e411ef85fef849e979cfed8/docs/index.md?plain=1#L86
 
- [ ] dataclass update the documentation for the dataclass, all functions should show in the mermaid overview: https://github.com/neurogeriatricskiel/KielMAT/blob/c366fd834ffe8fd60e411ef85fef849e979cfed8/docs/dataclass.md?plain=1#L6
 
- [ ] return the return for the turn detection detect function does not render correctly in the documentation: https://github.com/neurogeriatricskiel/KielMAT/blob/c366fd834ffe8fd60e411ef85fef849e979cfed8/docs/modules/td.md?plain=1#L3
 
- [ ] progressbar the documentation for the fetch_dataset function says there is a progressbar available, which is not in the documentation: https://github.com/neurogeriatricskiel/KielMAT/blob/c366fd834ffe8fd60e411ef85fef849e979cfed8/docs/datasets/keepcontrol.md?plain=1#L7
 
 all minor warning from the mkdocs autochcker: 
 - [ ] WARNING-  griffe: kielmat\utils\kielmat_dataclass.py:94: No type or annotation for returned value 'str'
 - [ ] WARNING -  griffe: kielmat\modules\ptd\_pham.py:90: Parameter 'min_turn_angle_deg' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\importers.py:19: No type or annotation for returned value 1
 - [ ] WARNING -  griffe: kielmat\utils\importers.py:76: Parameter 'tracked_point' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\quaternion.py:294: Parameter 'scalar_first' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\quaternion.py:296: Parameter 'channels_last' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:443: Parameter 'input_array' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:175: No type or annotation for parameter '**kwargs'
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:178: No type or annotation for returned value 1
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:85: Parameter 'param' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:1092: Parameter 'Data' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:1093: Parameter 'Window' does not appear in the function signature
 - [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:1096: No type or annotation for returned value 1

**Comment by masoudabedinifar** [comment]
Hi @JuliusWelzel ,

Thanks for the valuable feedback. I have tried to address all the concerns and have committed the revisions in 218949ff787000d38250dd2736cd09515cf4ac1d . Could you please check the revisions, and if they meet the requirements, merge the branch into the main branch?
Thanks!

Best, 
Masoud

File: kielmat/modules/ptd/_pham.py
Comment (JuliusWelzel) [inline kielmat/modules/ptd/_pham.py:1]: Does still not render properly: 
image

File: kielmat/modules/td/_pham.py
Comment (JuliusWelzel) [inline kielmat/modules/td/_pham.py:1]: Still not rendered properly: 
image

File: kielmat/utils/importers.py
Comment (JuliusWelzel) [inline kielmat/utils/importers.py:1]: You have to change the docstring of this function

File: kielmat/utils/kielmat_dataclass.py
Comment (JuliusWelzel) [inline kielmat/utils/kielmat_dataclass.py:1]: Not correct: 
image

File: mkdocs.yml
Comment (JuliusWelzel) [inline mkdocs.yml:1]: There should not be example with the same number. Maybe we can name them 1-8

**Comment by JuliusWelzel** [comment]
I also got more warning from mkdocs:

- [ ] WARNING -  griffe: kielmat\utils\kielmat_dataclass.py:94: No type or annotation for returned value 'str'
- [ ] WARNING -  griffe: kielmat\modules\ptd\_pham.py:90: Parameter 'min_turn_angle_deg' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\importers.py:19: No type or annotation for returned value 1
- [ ] WARNING -  griffe: kielmat\utils\importers.py:76: Parameter 'tracked_point' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\quaternion.py:294: Parameter 'scalar_first' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\quaternion.py:296: Parameter 'channels_last' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:443: Parameter 'input_array' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:175: No type or annotation for parameter '**kwargs'
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:178: No type or annotation for returned value 1
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:85: Parameter 'param' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:1092: Parameter 'Data' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:1093: Parameter 'Window' does not appear in the function signature
- [ ] WARNING -  griffe: kielmat\utils\preprocessing.py:1096: No type or annotation for returned value 1

**Comment by JuliusWelzel** [comment]
> Could you please check the revisions, and if they meet the requirements, merge the branch into the main branch?

For the future, please make sure you can run `mkdocs serve` and check the rendering yourself.

**Comment by masoudabedinifar** [comment]
> I also got more warning from mkdocs:
> 
> * [ ]  WARNING -  griffe: kielmat\utils\kielmat_dataclass.py:94: No type or annotation for returned value 'str'
> * [ ]  WARNING -  griffe: kielmat\modules\ptd_pham.py:90: Parameter 'min_turn_angle_deg' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\importers.py:19: No type or annotation for returned value 1
> * [ ]  WARNING -  griffe: kielmat\utils\importers.py:76: Parameter 'tracked_point' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\quaternion.py:294: Parameter 'scalar_first' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\quaternion.py:296: Parameter 'channels_last' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:443: Parameter 'input_array' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:175: No type or annotation for parameter '**kwargs'
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:178: No type or annotation for returned value 1
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:85: Parameter 'param' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:1092: Parameter 'Data' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:1093: Parameter 'Window' does not appear in the function signature
> * [ ]  WARNING -  griffe: kielmat\utils\preprocessing.py:1096: No type or annotation for returned value 1

I handled all these warnings and I do not get them when I render locally.

File: kielmat/datasets/keepcontrol.py
[the remark this answers is not in the document]
Comment (masoudabedinifar) [reply kielmat/datasets/keepcontrol.py:1]: It is handled here: 73bad3ef08997fa126ecdc8004da5814118267b9

File: kielmat/modules/ptd/_pham.py
Comment (masoudabedinifar) [reply kielmat/modules/ptd/_pham.py:1]: Handled in 5514104fc516e24cb0dfb3801b5aba3658ec194f

File: kielmat/modules/td/_pham.py
Comment (masoudabedinifar) [reply kielmat/modules/td/_pham.py:1]: Handled in 5514104fc516e24cb0dfb3801b5aba3658ec194f

File: kielmat/utils/importers.py
Comment (masoudabedinifar) [reply kielmat/utils/importers.py:1]: Handled here e6a4ab18a2975f67af059bf9cb795f25fa2bb281

File: kielmat/utils/kielmat_dataclass.py
Comment (masoudabedinifar) [reply kielmat/utils/kielmat_dataclass.py:1]: Handled here: 7026ba2a10a9544f14e40c84872946342fabce3c

File: mkdocs.yml
Comment (masoudabedinifar) [reply mkdocs.yml:1]: Handled here: 3a21b363d3e35f9d95f0b6e7b48f29c066d3566a

File: docs/dataclass.md
[the remark this answers is not in the document]
Comment (masoudabedinifar) [reply docs/dataclass.md:1]: Handled here: 0185add310749600b7ffa121e6c83413a1214164

**Comment by masoudabedinifar** [comment]
@JuliusWelzel Thanks for the feedback. I handled all the reviews and locally rendered them without any warnings.

**Review by JuliusWelzel — APPROVED** [review]
I am happy with the changes, maybe @rmndrs89 can also review. Once done, we are good to go

File: kielmat/utils/preprocessing.py
Comment (rmndrs89) [inline kielmat/utils/preprocessing.py:1]: Do `**kwargs` need to be specified as a dictionary? See [here](https://realpython.com/python-kwargs-and-args/)

**Review by rmndrs89 — APPROVED** [review]

**Pull request merged**